### PR TITLE
Replace libstd, libcore, liballoc terminology in docs

### LINKED ISSUE #103551: Replace libstd, libcore, liballoc terminology in docs
There are a few places in the standard library docs that still refer to libstd, libcore, and liballoc; the modern naming for these would be `std`, `core`, and `alloc`. We should update the docs.

```
$ rg 'libcore|libstd|liballoc' build/x86_64-unknown-linux-gnu/doc/ -l | grep -v '\.rs\.html'
build/x86_64-unknown-linux-gnu/doc/std/backtrace/index.html
build/x86_64-unknown-linux-gnu/doc/std/fmt/trait.Debug.html
build/x86_64-unknown-linux-gnu/doc/std/slice/trait.Concat.html
build/x86_64-unknown-linux-gnu/doc/alloc/index.html
build/x86_64-unknown-linux-gnu/doc/alloc/fmt/trait.Debug.html
build/x86_64-unknown-linux-gnu/doc/alloc/slice/trait.Concat.html
build/x86_64-unknown-linux-gnu/doc/core/sidebar-items1.66.0.js
build/x86_64-unknown-linux-gnu/doc/core/panicking/sidebar-items1.66.0.js
build/x86_64-unknown-linux-gnu/doc/core/panicking/fn.panic.html
build/x86_64-unknown-linux-gnu/doc/search-index1.66.0.js
build/x86_64-unknown-linux-gnu/doc/core/panicking/index.html
build/x86_64-unknown-linux-gnu/doc/core/fmt/trait.Debug.html
build/x86_64-unknown-linux-gnu/doc/core/prelude/index.html
build/x86_64-unknown-linux-gnu/doc/core/index.html
```

### PR DESCRIPTION
**Pull request by jonathanCogan**
Fixes #103551.  I changed line comments containing the outdated terms as well.

It would be great if someone with more experience could weigh in on whether these changes introduce ambiguity as suggested in https://github.com/rust-lang/rust/issues/103551#issuecomment-1291225315.

### CODE REVIEW

File: library/alloc/src/slice.rs [hunk 1]
```diff
@@ -653,7 +653,7 @@ impl [u8] {
 ///
 /// ```error
 /// error[E0207]: the type parameter `T` is not constrained by the impl trait, self type, or predica
-///    --> src/liballoc/slice.rs:608:6
+///    --> src/alloc/slice.rs:608:6
```
Comment (H4x5) [inline library/alloc/src/slice.rs:656]: The correct path here is actually `library/alloc/src/slice.rs`

File: library/core/tests/str.rs [hunk 2]
```diff
@@ -1 +1 @@
-// All `str` tests live in liballoc/tests
+// All `str` tests live in alloc/tests
```
Comment (H4x5) [inline library/core/tests/str.rs:1]: Maybe this should be `library/alloc/tests/str.rs`?

File: library/core/src/lib.rs [hunk 3]
```diff
@@ -38,18 +38,18 @@
 //!    which do not trigger a panic can be assured that this function is never
 //!    called. The `lang` attribute is called `eh_personality`.
 
-// Since libcore defines many fundamental lang items, all tests live in a
+// Since core defines many fundamental lang items, all tests live in a
 // separate crate, libcoretest, to avoid bizarre issues.
```
Comment (H4x5) [inline library/core/src/lib.rs:42]: the "libcoretest" crate lives at `library/core/tests`, maybe mention that here?

**Review by ghost — CHANGES_REQUESTED** [review]

**Review by ghost — APPROVED** [review]

**Pull request merged**